tests/Tests/ORM/Functional/SecondLevelCacheQueryCacheTest.php

```diff
@@ -939,7 +938,6 @@ public function testResolveAssociationCacheEntry(): void
         self::assertNotNull($state1->getCountry());
         $this->assertQueryCount(1);
         self::assertInstanceOf(State::class, $state1);
-        self::assertInstanceOf(InternalProxy::class, $state1->getCountry());
```

Comment:
It would be useful to have a `BaseTestCase::isLazyObject` function for this assertion,  I did one for [MongoDB ODM](https://github.com/doctrine/mongodb-odm/pull/2700/files#r1855280020).